src/AttachmentAction.php

```diff
@@ -179,13 +206,21 @@ public function setConfirm(?ActionConfirmation $confirm): self
      */
     public function toArray(): array
     {
-        return [
+        $array = [
             'name' => $this->name,
             'text' => $this->text,
             'style' => $this->style,
             'type' => $this->type,
             'value' => $this->value,
             'confirm' => $this->confirm ? $this->confirm->toArray() : null,
         ];
+
+        // link buttons do not require "name", "value" and "confirm" attributes
+        if (null !== $this->url) {
+            unset($array['name'], $array['value'], $array['confirm']);
```

Comment:
Why not start initialization without those attribute and set them after of the URL is not provided.